spec/vector_spec.rb

```diff
@@ -254,6 +254,115 @@
             expect { @vector[:x, :one] }.to raise_error(IndexError)
           end
         end
+
+        context Daru::CategoricalIndex do
+          context "non-numerical index" do
+            let (:idx) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+            let (:dv)  { Daru::Vector.new 'a'..'e', index: idx1 }
+
+            context "single category" do
+              context "multiple instances" do
+                subject { dv[:a] }
+
+                it { is_expected.to be_a Daru::Vector }
+                its(:size) { is_expected.to eq 3 }
+                its(:to_a) { is_expected.to eq  ['a', 'c', 'd'] }
+                its(:index) { is_expected.to eq(
+                  Daru::CategoricalIndex.new([:a, :a, :a])) }
+              end
+
+              context "single instance" do
+                subject { dv[:c] }
```

Comment:
`let` statements are re-executed for each example (`it`), and only when `let`-ed thing is called (otherwise, they are not executed at all).